core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -328,11 +338,23 @@ private void createDatasets(@Nullable PipelineConfigurer pipelineConfigurer, @Nu
 
       if (!excelInputreaderConfig.containsMacro("memoryTableName") &&
         !Strings.isNullOrEmpty(excelInputreaderConfig.memoryTableName)) {
+
+        ExploreProperties.Builder builder = ExploreProperties.builder();
+        if (!Strings.isNullOrEmpty(excelInputreaderConfig.memoryDatasetExplorer)) {
```

Comment:
this logic is repeated multiple times. use defaults in the config constructors to remove this